### PR TITLE
test: add seeded tests for random module

### LINKED ISSUE #809: Add seeded runs for random module
In most our tests we have a section for seeded based tests that tests the deterministic of a module.
Somehow we currently don't have that for the random module. So let's add them.
Contributor can orientate on other test files.

### PR DESCRIPTION
**Pull request by ST-DDT**
Supersedes #837
Fixes #809

This PR is way easier to review if you omit white-space changes.

### CODE REVIEW

**Review by Shinigami92 — APPROVED** [review]

File: test/random.spec.ts [hunk 1]
```diff
@@ -2,328 +2,429 @@ import { beforeEach, describe, expect, it, vi } from 'vitest';
...
+        it('should be able to ban all numeric characters', () => {
+          const bannedChars = '0123456789'.split('');
+          const alphaText = faker.random.alphaNumeric(5, {
+            bannedChars,
+          });
+
+          expect(alphaText).toHaveLength(5);
+          for (const bannedChar of bannedChars) {
+            expect(alphaText).not.includes(bannedChar);
+          }
```
Comment (import-brain) [inline test/random.spec.ts:373]:
```suggestion
          expect(alphaText).toMatch(/^([^0-9]*)$/);
```

Comment (import-brain) [reply test/random.spec.ts:373]: If you're testing to make sure that no numeric characters make it into the output, can't this be used? This regex matches all characters except numbers.

Comment (ST-DDT) [reply test/random.spec.ts:373]: I haven't changed these lines. I only changed lines above ~L100.
Please turn off white-space changes.
If you think, an existing test needs to be updated, please open a separate issue/PR for that.

Comment (import-brain) [reply test/random.spec.ts:377]: Whoops, forgot to turn off whitespace changes, that's my bad. I'll open a new PR.

**Review by Shinigami92 — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Pull request merged**